pennylane/transforms/qcut.py

```diff
@@ -939,6 +940,142 @@ def qcut_processing_fn_mc(
+    measurement. When decorated with ``@qml.cut_circuit_mc``, we can cut the circuit into two
+    :math:`2`-qubit fragments:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.cut_circuit_mc(shots=shots, device_wires=[0, 1])
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(0.89, wires=0)
+            qml.RY(0.5, wires=1)
+            qml.RX(1.3, wires=2)
+
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=[1, 2])
+
+            qml.RX(0.3, wires=0)
+            qml.RY(0.7, wires=1)
+            qml.RX(2.3, wires=2)
+            return qml.sample(wires=[0, 1, 2])
+
+    TODO: finish example with working code, add usage details
+
+    """
+
+    g = tape_to_graph(tape)
+    replace_wire_cut_nodes(g)
+    fragments, communication_graph = fragment_graph(g)
+    fragment_tapes = [graph_to_tape(f) for f in fragments]
+    fragment_tapes = [remap_tape_wires(t, device_wires) for t in fragment_tapes]
+
+    configurations, settings = expand_fragment_tapes_mc(
+        fragment_tapes, communication_graph, shots=shots
+    )
+
+    tapes = tuple(tape for c in configurations for tape in c)
+
+    if classical_processing_fn:
+        return tapes, partial(
+            qcut_processing_fn_mc,
+            communication_graph=communication_graph,
+            settings=settings,
+            shots=shots,
+            classical_processing_fn=classical_processing_fn,
+        )
+    else:
+        return tapes, partial(
+            qcut_processing_fn_sample, communication_graph=communication_graph, shots=shots
+        )
+
+
+@cut_circuit_mc.custom_qnode_wrapper
+def qnode_execution_wrapper(self, qnode, targs, tkwargs):
+    """Here, we overwrite the QNode execution wrapper in order
+    to replace execution variables"""
+
+    transform_max_diff = tkwargs.pop("max_diff", None)
+    tkwargs.setdefault("device_wires", qnode.device.wires)
+
+    if "shots" in inspect.signature(qnode.func).parameters:
+        raise ValueError(
+            "Detected 'shots' as an argument of the quantum function to transform. "
+            "The 'shots' argument name is reserved for overriding the number of shots "
+            "taken by the device."
+        )
+
+    def _wrapper(*args, **kwargs):
+        qnode.construct(args, kwargs)
+        tapes, processing_fn = self.construct(qnode.qtape, *targs, **tkwargs)
+
+        interface = qnode.interface
+        execute_kwargs = getattr(qnode, "execute_kwargs", {}).copy()
+        max_diff = execute_kwargs.pop("max_diff", 2)
+        max_diff = transform_max_diff or max_diff
+
+        gradient_fn = getattr(qnode, "gradient_fn", qnode.diff_method)
+        gradient_kwargs = getattr(qnode, "gradient_kwargs", {})
+
+        if interface is None or not self.differentiable:
+            gradient_fn = None
+
+        execute_kwargs["cache"] = False
```

Comment:
```suggestion
        if execute_kwargs.get("cache", False):
            warnings.warn("Sample-based circuit cutting only supports setting "
                                    "cache=False when instantiating the QNode. Continuing "
                                    "evalutaion without using a cache.", UserWarning)
        execute_kwargs["cache"] = False
```
I'm not sure about this, it means we'll raise a warning by default since caching is on by default. Though I still think we should discuss with the rest of the PL team whether caching should be used at all when returning samples :thinking: